### PR TITLE
Feature/operations whitelisting

### LINKED ISSUE #151: Make introspection configurable
Introspection should be an environment variable, defaulting to ~~`true`~~`false`
https://github.com/input-output-hk/cardano-graphql/blob/master/src/Server.ts#L26

### PR DESCRIPTION
**Pull request by rhyslbw**
Replaces primary intent of #163, by implementing a white listing feature in the public-facing layer,. This removes the [upstream blocker](https://github.com/hasura/graphql-engine/issues/4687), and improves the design since the API can then be extended with data sources other than Hasura.

Also Closes #151

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -28,35 +32,44 @@ export function getConfig (): Config {
   if (prometheusMetrics && process.env.TRACING === 'false') {
     throw new TracingRequired('Prometheus')
   }
+  if (whitelistPath && allowIntrospection) {
+    throw new IntrospectionNotPermitted('whitelist')
+  }
```
Comment (KtorZ) [inline src/config.ts:37]: So we can't have whitelisting and allowIntrospection at the same time is forbidden. Introspection was allowed by default before. So what about simply having `allowIntrospection = !whitelistPath` to keep the previous behavior roughly the same and, not bother with this error scenario at all ?

Comment (rhyslbw) [reply src/config.ts:37]: I feel the previous default was wrong, and introspection should be opt-in

Comment (rhyslbw) [reply src/config.ts:37]: ApolloServer defaults to `true` when `process.env.NODE_ENV` is not `production`, but I just find these environmental variances cause more confusion than they solve.

File: src/__test__/Server.test.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        console.error(error)
+        return
+      } finally {
+        httpServer.close()
+      }
+    })
+
+    describe('Providing a whitelist produced by persistgraphql, intended to lock the API for specific applications', () => {
+      beforeEach(async () => {
+        proc = exec(`npx persistgraphql ${clientPath} ${whitelistPath}`)
```
Comment (rhyslbw) [inline src/__test__/Server.test.ts:86]: @disassembler 
`npx persistgraphql ${clientPath} ${whitelistPath}` creates the whitelist

File: src/config.ts [hunk 3]
```diff
@@ -28,35 +32,44 @@ export function getConfig (): Config {
...
+    ALLOW_INTROSPECTION,
     ALLOWED_ORIGINS,
     API_PORT,
     CACHE_ENABLED,
     HASURA_URI,
     PROMETHEUS_METRICS,
     QUERY_DEPTH_LIMIT,
-    TRACING
+    TRACING,
+    WHITELIST_PATH
```
Comment (rhyslbw) [inline src/config.ts:62]: @disassembler 
This points to just the root dir, without the glob pattern

Comment (rhyslbw) [inline src/config.ts:62]: configure here

**Pull request merged**